tests/analyzer/tests.py

```diff
@@ -195,1279 +197,1343 @@ def test_detect_analyzer_gradle(self):
         log = '189738938-orig.log'
         job_id = 189738938
         file_path = 'python/' + log
         python20 = self.dispatcher.analyze(file_path, job_id)
-        yield self.compare_analyzer, python20, 'python'
-        yield self.compare_num_t_run, python20, 7607
-        yield self.compare_num_t_ok, python20, 7606
-        yield self.compare_num_t_failed, python20, 1
-        yield self.compare_num_t_skipped, python20, 62
-        yield self.compare_bool_t_ran, python20, True
-        yield self.compare_bool_t_failed, python20, True
-        yield self.compare_frameworks, python20, 'unittest'
-        yield self.compare_tr_t_failed, python20, "sklearn.neighbors.classification.KNeighborsClassifier"
+        self.compare_analyzer(python20, 'python')
+        self.compare_num_t_run(python20, 7607)
+        self.compare_num_t_ok(python20, 7606)
+        self.compare_num_t_failed(python20, 1)
+        self.compare_num_t_skipped(python20, 62)
+        self.compare_bool_t_ran(python20, True)
+        self.compare_bool_t_failed(python20, True)
+        self.compare_frameworks(python20, 'unittest')
+        self.compare_tr_t_failed(python20, "sklearn.neighbors.classification.KNeighborsClassifier")
 
     def test_gradle_0(self):
         log = "88551599-orig.log"
         job_id = 88551599
         file_path = "gradle/" + log
         ts, r = self.get_trigger_sha_and_repo(job_id)
         gradle0 = self.dispatcher.analyze(file_path, job_id, trigger_sha=ts, repo=r)
-        yield self.compare_status, gradle0, "broken"
-        yield self.compare_analyzer, gradle0, "java-gradle"
-        yield self.compare_num_t_run, gradle0, 1160
-        yield self.compare_num_t_ok, gradle0, 1118
-        yield self.compare_num_t_failed, gradle0, 42
-        yield self.compare_num_t_skipped, gradle0, 0
-        yield self.compare_bool_t_ran, gradle0, True
-        yield self.compare_bool_t_failed, gradle0, True
+        self.compare_status(gradle0, "broken")
+        self.compare_analyzer(gradle0, "java-gradle")
+        self.compare_num_t_run(gradle0, 1160)
+        self.compare_num_t_ok(gradle0, 1118)
+        self.compare_num_t_failed(gradle0, 42)
+        self.compare_num_t_skipped(gradle0, 0)
+        self.compare_bool_t_ran(gradle0, True)
+        self.compare_bool_t_failed(gradle0, True)
 
     def test_gradle_1(self):
         log = "68605615-orig.log"
         job_id = 68605615
         file_path = "gradle/" + log
         ts, r = self.get_trigger_sha_and_repo(job_id)
         gradle1 = self.dispatcher.analyze(file_path, job_id, trigger_sha=ts, repo=r)
-        yield self.compare_status, gradle1, "broken"
-        yield self.compare_analyzer, gradle1, "java-gradle"
-        yield self.compare_num_t_run, gradle1, 1143
-        yield self.compare_num_t_ok, gradle1, 1142
-        yield self.compare_num_t_failed, gradle1, 1
-        yield self.compare_num_t_skipped, gradle1, 0
-        yield self.compare_bool_t_ran, gradle1, True
-        yield self.comp
```

Comment:
I'm not sure if I understand this correctly, does `compare_single_log` return a boolean? I'm assuming you're saying that it's `True` meaning the reproduced and original log match?

Reply:
`compare_single_log` returns a tuple. The first element is a boolean that is `True` whether the reproduced and original logs match, and `False` otherwise. The second element is a list of dicts describing in what respects the reproduced and original logs differ.